internal/webhook/external/github.go

```diff
@@ -153,7 +157,41 @@ func (g *githubWebhookReceiver) GetHandler() http.HandlerFunc {
 			return
 		}
 
+		var repoURL string
+
 		switch e := event.(type) {
+		case *gh.PackageEvent:
+			switch e.GetAction() {
+			// These are the only actions that should refresh Warehouses.
+			case "published", "updated":
+			default:
+				xhttp.WriteResponseJSON(w, http.StatusOK, nil)
+				return
+			}
+			pkg := e.GetPackage()
+			if pkg == nil || pkg.GetPackageVersion() == nil {
+				xhttp.WriteErrorJSON(
+					w,
+					xhttp.Error(errors.New("invalid request body"), http.StatusBadRequest),
+				)
+				return
+			}
+			switch pkg.GetPackageType() {
+			// These are the only types of packages we care about.
+			case ghcrPackageTypeContainer, ghcrPackageTypeDocker:
+			default:
+				xhttp.WriteResponseJSON(w, http.StatusOK, nil)
+				return
+			}
+			var ref name.Reference
+			if ref, err = name.ParseReference(
+				*pkg.GetPackageVersion().PackageURL,
```

Comment:
If we use the `GetX` methods that Github provides we don't have to dereference + we get the nil check for free

```suggestion
				pkg.GetPackageVersion().GetPackageURL(),
```

Reply:
Nice catch @fuskovic. Agree.